src/unix/stream.c

```diff
@@ -852,14 +859,7 @@ static void uv__write(uv_stream_t* stream) {
 
   if (n < 0) {
     if (errno != EAGAIN && errno != EWOULDBLOCK) {
-      /* Error */
```

Comment:
here you would just assign that variable to -errno and do the goto as you do now.